### PR TITLE
Big decimal

### PR DESCRIPTION
**Pull request by leodasvacas**
Wip integration of big decimal.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -503,7 +557,7 @@ export class EthereumValue {
 export enum ValueKind {
   STRING = 0,
   INT = 1,
-  FLOAT = 2,
```
Comment (Jannis) [inline index.ts]: Another breaking change, although this one is more behind the scenes, so I'd be ok with it. We did say we'd replace `Float` with `BigDecimal`.

File: index.ts [hunk 2]
```diff
@@ -214,6 +225,49 @@ export class BigInt extends Uint8Array {
   }
```
Comment (Jannis) [inline index.ts]: Would it make sense to add `toBigDecimal()` to `BigInt` that uses the `bigDecimal.fromBigInt(x: BigInt): BigDecimal` that I proposed above?

Comment (leodasvacas) [reply index.ts]: Done.

File: index.ts [hunk 3]
```diff
@@ -53,10 +53,21 @@ declare namespace bigInt {
   function plus(x: BigInt, y: BigInt): BigInt
   function minus(x: BigInt, y: BigInt): BigInt
   function times(x: BigInt, y: BigInt): BigInt
-  function dividedBy(x: BigInt, y: BigInt): BigInt
```
[the remark this answers is not in the document]
Comment (leodasvacas) [reply index.ts]: Keeping integer division by default seems reasonable.

File: index.ts [hunk 4]
```diff
@@ -53,10 +53,21 @@ declare namespace bigInt {
...
+/** Host interface for BigDecimal */
+declare namespace bigDecimal {
+  function plus(x: BigDecimal, y: BigDecimal): BigDecimal
+  function minus(x: BigDecimal, y: BigDecimal): BigDecimal
+  function times(x: BigDecimal, y: BigDecimal): BigDecimal
+  function dividedBy(x: BigDecimal, y: BigDecimal): BigDecimal
+  function equals(x: BigDecimal, y: BigDecimal): boolean
+  function toString(bigDecimal: BigDecimal): string
+  function fromString(s: string): BigDecimal
+}
```
[the remark this answers is not in the document]
Comment (leodasvacas) [reply index.ts]: `fromBigInt(x)` is `new BigDecimal(x)`. I agree those operations are important but they're not readily available in the Rust BigDecimal library, we could probably implement them in AS but it's not so obvious so lets leave it for another day https://github.com/graphprotocol/graph-ts/issues/48.

**Comment by leodasvacas** [comment]
@Jannis review addressed.

File: index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jannis) [reply index.ts]: Ok, agreed!

**Pull request merged**